drivers/spi/spi_ambiq.c

```diff
@@ -0,0 +1,238 @@
+		ret = am_hal_iom_spi_blocking_fullduplex(data->IOMHandle, &trans);
+
+		memcpy(ctx->rx_buf, &buf3, (ctx->rx_len * sizeof(uint8_t)));
+
+	} else if (ctx->tx_buf != NULL) {
+		spi_context_update_tx(ctx, 1, 1);
+		trans.ui32NumBytes = ctx->tx_len;
+		trans.pui32TxBuffer = (uint32_t *)ctx->tx_buf;
+		trans.pui32RxBuffer = (uint32_t *)ctx->tx_buf;
+		ret = am_hal_iom_spi_blocking_fullduplex(data->IOMHandle, &trans);
+	}
+
+	spi_context_complete(ctx, dev, 0);
+
+	return ret;
+}
+
+static int spi_ambiq_transceive(const struct device *dev, const struct spi_config *config,
+				const struct spi_buf_set *tx_bufs,
+				const struct spi_buf_set *rx_bufs)
+{
+	struct spi_ambiq_data *data = dev->data;
+	int ret;
+
+	ret = spi_config(dev, config);
+
+	if (ret) {
+		return ret;
+	}
+
+	if (!tx_bufs && !rx_bufs) {
+		return 0;
+	}
+
+	spi_context_buffers_setup(&data->ctx, tx_bufs, rx_bufs, 1);
+
+	ret = spi_ambiq_xfer(dev, config);
+
+	return ret;
+}
+
+static int spi_ambiq_release(const struct device *dev, const struct spi_config *config)
+{
+	struct spi_ambiq_data *data = dev->data;
+
+	if (!sys_read32(SPI_STAT(dev))) {
+		return -EBUSY;
+	}
+
+	spi_context_unlock_unconditionally(&data->ctx);
+
+	return 0;
+}
+
+static struct spi_driver_api spi_ambiq_driver_api = {
+	.transceive = spi_ambiq_transceive,
+	.release = spi_ambiq_release,
+};
+
+static int spi_ambiq_init(const struct device *dev)
+{
+	struct spi_ambiq_data *data = dev->data;
+	const struct spi_ambiq_config *cfg = dev->config;
+	int ret;
+
+	ret = am_hal_iom_initialize((cfg->base - REG_IOM_BASEADDR) / cfg->size, &data->IOMHandle);
+
+	ret = cfg->pwr_func();
+
+	ret = pinctrl_apply_state(cfg->pcfg, PINCTRL_STATE_DEFAULT);
+
+	return ret;
+}
+
+#define AMBIQ_SPI_INIT(n)                                                                          \
+	PINCTRL_DT_INST_DEFINE(n);                                                                 \
+	static int pwr_on_ambiq_spi_##n(void)                                                      \
+	{                                                                                          \
+		uint32_t addr = DT_REG_ADDR(DT_INST_PHANDLE(n, ambiq_pwrcfg)) +                    \
+				DT_INST_PHA(n, ambiq_pwrcfg, offset);                              \
+		sys_write32((sys_read32(addr) | DT_INST_PHA(n, ambiq_pwrcfg, mask)), addr);        \
+		k_busy_wait(PWRCTRL_MAX_WAIT_US);                                                  \
+		return 0;                                                                          \
+	}                                                                                          \
+	static struct spi_ambiq_data spi_ambiq_data##n = {                                         \
+		SPI_CONTEXT_INIT_LOCK(spi_ambiq_data##n, ctx),                                     \
+		SPI_CONTEXT_INIT_SYNC(spi_ambiq_data##n, ctx),                                     \
+		SPI_CONTEXT_CS_GPIOS_INITIALIZE(DT_DRV_INST(n), ctx)};                             \
```

Comment:
You also need to call `spi_context_cs_configure_all` function during the driver initialization + `spi_context_cs_control` during transceiving.

Reply:
The `SPI_CONTEXT_CS_GPIOS_INITIALIZE` was a remnant. 
This driver will not support this functionality. CS is handled by HAL functions and can be selected by `ambiq,iom-nce-module` in pinctrl.

Reply:
This approach forces to use the dedicated CS gpios only. Why don't you follow the way as it is done for the rest SPI drivers in Zephyr tree?

Reply:
@bbilas is this really a blocker? You can still configure the CS pins via pinctrl configuration and it will work the same way.